packages/website/docs/sending-algolia-insights-events.md

```diff
@@ -0,0 +1,223 @@
+
+## Customizing events
+
+You can change any of the plugin's default behavior by using the [`onItemsChange`](createAlgoliaInsightsPlugin#onitemschange), [`onSelect`](createAlgoliaInsightsPlugin#onselect), or [`onActive`](createAlgoliaInsightsPlugin#onactive) hooks. For example, you may want to customize the `eventName` to match your [naming conventions](https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-and-conversion-analytics/in-depth/best-practices-for-sending-events/#naming-events-consistently) or send a different events on these actions.
+
+This snippet shows how to instantiate a plugin that sends a click event with "Product Selected from Autocomplete" as the `eventName` whenever a user selects an item.
+
+```js title="index.js"
+const appId = "latency";
+const apiKey = "6be0576ff61c053d5f9a3225e2a90f76";
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient("init", { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
+  insightsClient,
+  onSelect({ insights, insightsEvents }) {
+    const events = insightsEvents.map((insightsEvent) => ({
+      ...insightsEvent,
+      eventName: 'Product Selected from Autocomplete',
+    }));
+    insights.clickedObjectIDsAfterSearch(...events);
+  },
+})
+```
+
+If you're [using multiple different Algolia indices in the same autocomplete](adding-multiple-result-types), for example one for products and one for suggestions, you may want to use different `eventNames` for each section. You can do this conditionally based on the index name:
+
+```js title="index.js"
+const appId = "latency";
+const apiKey = "6be0576ff61c053d5f9a3225e2a90f76";
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient("init", { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
+  insightsClient,
+  onSelect({ insights, insightsEvents }) {
+    const events = insightsEvents.map((insightsEvent) => {
+      switch (insightsEvent.index) {
+        case 'instant_search_demo_query_suggestions': {
+          return {
+            ...insightsEvent,
+            eventName: 'Suggestion Selected from Autocomplete',
+          };
+        }
+        case 'instant_search': {
+          return {
+            ...insightsEvent,
+            eventName: 'Product Selected from Autocomplete',
+          };
+        }
+        default: {
+          return {
+            ...insightsEvent,
+            eventName: 'Item Selected from Autocomplete',
+          };
+        }
+      }
+    });
+    insights.clickedObjectIDsAfterSearch(...events);
+  },
+});
+```
+
```

Comment:
We should add one more section here, which could be called something like "Sending events from user interactions". If you want to send custom events from your templates (e.g., a conversion event from an "Add to cart" button), the Insights API is stored in the Context.

I created [an example in the playground](https://github.com/algolia/autocomplete.js/blob/3bc61d506a1fdde4091339ad91286cf74b9425b3/examples/js/app.tsx#L156-L161).

Reply:
Great idea! I've added a new section called "Sending events from templates" since the other sections could also be described by "Sending events from user actions".

Hope this title and section is what you were thinking of.

I've also moved this guide * after * the adding suggested searches, recent searches, and multiple results types guides since I think it makes more sense to learn this info after setting up your sources.